libclc/ptx-nvidiacl/libspirv/async/wait_group_events.cl

```diff
@@ -0,0 +1,21 @@
+
+//===----------------------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include <spirv/spirv.h>
+
+int __clc_nvvm_reflect_arch();
+
+_CLC_OVERLOAD _CLC_DEF void __spirv_GroupWaitEvents(unsigned int scope,
+                                                    int num_events,
+                                                    event_t *event_list) {
+  if (__clc_nvvm_reflect_arch() >= 800) {
+    __nvvm_cp_async_wait_all();
+  }
+  __spirv_ControlBarrier(scope, scope, SequentiallyConsistent);
```

Comment:
I think `SequentiallyConsistent` is the right thing for this PR, because the SYCL 2020 specification is unclear about what the synchronization entails here.  But I wanted to call this out as something for us to remember for https://github.com/intel/llvm/pull/4950; the extension should either say what the memory order associated with the synchronization is supposed to be, or we should consider exposing it as part of the API.